analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Branching.cs

```diff
@@ -535,5 +535,84 @@ public void Branching_NotNullConstraints_VisitsElseBranch()
                 "Else",
                 "End");
         }
+
+        [DataTestMethod]
+        [DynamicData(nameof(StringIsNullOrEmptyMethods))]
+        public void Branching_IsNullOrEmpty_SetsNullConstraintsInBranches(string methodName)
+        {
+            var code = $@"
+if (string.{methodName}(s))
+{{
+    Tag(""If"", s);
+}}
+else
+{{
+    Tag(""Else"", s);
+}}
+Tag(""End"", s);";
+            var validator = SETestContext.CreateCS(code, ", string s").Validator;
+            validator.TagValues("If").Should().HaveCount(2)
+                .And.ContainSingle(x => x.HasConstraint(ObjectConstraint.Null))
+                .And.ContainSingle(x => x.HasConstraint(ObjectConstraint.NotNull));
+            validator.ValidateTag("Else", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue());
+            validator.TagValues("End").Should().HaveCount(2)
+                .And.ContainSingle(x => x.HasConstraint(ObjectConstraint.Null))
+                .And.ContainSingle(x => x.HasConstraint(ObjectConstraint.NotNull));
+        }
+
+        [DataTestMethod]
+        [DynamicData(nameof(StringIsNullOrEmptyMethods))]
+        public void Branching_IsNullOrEmpty_VisitsTrueBranchIfNull(string methodName)
+        {
+            var code = $@"
+string s = null;
```

Comment:
Makes it easier to track
```suggestion
string isNull = null;
```